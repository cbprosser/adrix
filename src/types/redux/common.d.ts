export type RequestStatus =
  | {
      message?: string;
      requestId?: string;
      status: 'idle' | 'calling';
      success?: boolean;
    }
  | {
      message: string;
      requestId?: string;
      status: 'idle' | 'calling';
      success: false;
    };

export type APISlice<T extends string = string> = {
  requests: Record<T, RequestStatus>;
};
