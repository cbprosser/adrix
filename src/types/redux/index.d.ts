export * from './slices';

export * from './Root';
