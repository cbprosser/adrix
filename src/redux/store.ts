import { configureStore } from '@reduxjs/toolkit';
import { cardsSlice } from './slices/cards';

export const store = configureStore({
  reducer: {
    cards: cardsSlice.reducer,
  },
});
