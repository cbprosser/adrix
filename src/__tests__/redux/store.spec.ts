import { store } from '../../redux';

describe('store Suite', () => {
  it('Should have the correct properties', () => {
    expect(store.getState()).toMatchSnapshot();
  });
});
