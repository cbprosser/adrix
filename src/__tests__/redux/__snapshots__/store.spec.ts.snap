// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store Suite Should have the correct properties 1`] = `
Object {
  "cards": Object {
    "cards": Array [],
    "requests": Object {
      "loadSearch": Object {
        "status": "idle",
      },
      "searchForCard": Object {
        "status": "idle",
      },
    },
  },
}
`;
